src/core/components/sidepanels/panels/outline/arborist/ListTree.tsx

```diff
@@ -1,49 +1,138 @@
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSetHighlighted = useCallback(debounce(setHighlighted, 1000), []);
+
+  const handleToggle = (event: any) => {
+    event.stopPropagation();
+    /**
+     * Toggle the node open and close State
+     */
+    node.toggle();
+  };
+
+  /**
+   * To stop propagation of the event to the parent
+   * Tree Component to avoid clearing the selection of blocks
+   * and allowing to select other blocks
+   */
+  const handleClearSelection = (e) => {
+    e.stopPropagation();
+    /**
+     * It will work when a node is clicked.
+     * The onSelect in the parent Tree Component
+     * will also trigger the selection of the node.
+     */
+    handleClick(e);
+  };
+
   return (
-    <div style={style} ref={dragHandle}>
-      {node.isLeaf ? "🍁" : "🗀"}
-      {node.data.name}
+    <div
+      onClick={handleClearSelection}
+      key={id}
+      onMouseEnter={() => debouncedSetHighlighted(id)}
+      style={style}
+      ref={dragHandle}
+      className={cn(
+        "group flex !h-fit w-full items-center justify-between space-x-px py-px",
+         isSelected ? "bg-blue-500 text-white" : "text-gray-600 hover:bg-gray-200 dark:hover:bg-gray-800",
+      )}>
+      <div className="flex items-center">
+        <div
+          className={`flex h-4 w-4 rotate-0 transform cursor-pointer items-center justify-center text-xs transition-transform duration-100 ${
+            node.isOpen ? "rotate-90" : ""
+          }`}>
+          {node.children.length > 0 && (
+            <button onClick={handleToggle} type="button">
+              <TriangleRightIcon />
+            </button>
+          )}
+        </div>
+        <button type="button" className="flex items-center">
+          <div className="-mt-1 h-3 w-3">
+            <TypeIcon type={data?._type} />
+          </div>
+          <div className="ml-2 truncate text-[11px]">{data?._type}</div>
+        </button>
+      </div>
+      <div className="invisible flex items-center space-x-1 pr-2 group-hover:visible">
+        {outlineItems.map((outlineItem) => (
+          <Tooltip>
+            <TooltipTrigger
+              className="cursor-pointer rounded bg-transparent hover:bg-white hover:text-blue-500"
+              asChild>
+              {outlineItem.item(id)}
+            </TooltipTrigger>
+            <TooltipContent className="">{outlineItem.tooltip}</TooltipContent>
+          </Tooltip>
+        ))}
+      </div>
     </div>
   );
 }
 
 const ListTree = () => {
+  const [allBlocks] = useBlocksStore();
+
   const [ids, setIds] = useSelectedBlockIds();
   const [, setStyleBlocks] = useSelectedStylingBlocks();
+
   const clearSelection = () => {
     setIds([]);
     setStyleBlocks([]);
   };
 
+  const treeData = useMemo(() => {
+    let treeBlocks = getBlocksTree(allBlocks);
+    /***
+     * Removing the duplicates from the TreeBlocks
+     */
+    treeBlocks = removeDuplicates(treeBlocks);
```

Comment:
Not sure how this works, but seem like a hack. Lets keep it for now, but should revisit it again to see how we can get rid of this.

Reply:
It might be an issue with flatToNested library. If you want to you can try by writing your own function for the flat to nested conversion.

Reply:
Yeah. I was looking into the implementation and found that FlatToNested Library has some issues. So I decided to use that duplicate removal for now. I'll add a :FIXME in the code comment to come back Later.